lib/dao.js

```diff
@@ -58,11 +58,12 @@ function copyData(from, to) {
   }
 }
 
-function convertDataPropertiesByType(data) {
+function convertDataPropertiesByType(inst, data) {
   var typedData = {};
   for (var key in data) {
     // Convert the properties by type
-    typedData[key] = data[key];
+    inst[key] = data[key];
```

Comment:
I am confused. I though `inst` was updated by calling `copyData` just before calling `convertDataPropertiesByType`? If we are updating the instance as part of this method, then 1) do we need `copyData` at all? 2) the name of this method should be changed to better communicate the intent.